mlrun/feature_store/steps.py

```diff
@@ -204,13 +216,45 @@ def _do_pandas(self, event):
             elif feature_map:
                 # create and apply simple map
                 df[self._get_feature_name(feature)] = event[feature].map(
-                    lambda x: feature_map[x]
+                    lambda x: feature_map.get(x, None)
                 )
 
         if self.with_original_features:
             df = pd.concat([event, df], axis=1)
         return df
 
+    def _do_spark(self, event):
+        from itertools import chain
+
+        from pyspark.sql.functions import col, create_map, lit, when
+
+        for column, column_map in self.mapping.items():
+            new_column_name = self._get_feature_name(column)
+            if "ranges" not in column_map:
+                mapping_expr = create_map([lit(x) for x in chain(*column_map.items())])
+                event = event.withColumn(
+                    new_column_name, mapping_expr.getItem(col(column))
+                )
+            else:
+                for val, val_range in column_map["ranges"].items():
+                    min_val = val_range[0] if val_range[0] != "-inf" else -np.inf
+                    max_val = val_range[1] if val_range[1] != "inf" else np.inf
+                    otherwise = ""
+                    if new_column_name in event.columns:
+                        otherwise = event[new_column_name]
+                    event = event.withColumn(
+                        new_column_name,
+                        when(
+                            (event[column] < max_val) & (event[column] > min_val),
```

Comment:
```suggestion
                            (event[column] < max_val) & (event[column] >= min_val),
```
Ranges need to be `[min, max)`, otherwise you have gaps in the ranges.